x-pack/plugins/apm/public/components/shared/charts/timeline/timeline_axis.tsx

```diff
@@ -47,10 +45,20 @@ export function TimelineAxis({
   topTraceDuration,
 }: TimelineAxisProps) {
   const theme = useTheme();
-  const { margins, tickValues, width, xDomain, xMax, xScale } = plotValues;
+  const { margins, tickValues, width, xMax, xScale } = plotValues;
   const tickFormatter = getDurationFormatter(xMax);
-  const xAxisTickValues = getXAxisTickValues(tickValues, topTraceDuration);
-  const topTraceDurationFormatted = tickFormatter(topTraceDuration).formatted;
+
+  const tickPositionsAndLabels = getXAxisTickValues(
+    tickValues,
+    topTraceDuration
+  ).reduce<Array<{ position: number; label: string }>>((ticks, tick) => {
+    const position = xScale(tick);
+    return Number.isFinite(position)
+      ? [...ticks, { position, label: tickFormatter(tick).formatted }]
+      : ticks;
+  }, []);
+  const topTraceDurationPosition =
+    topTraceDuration > 0 ? xScale(topTraceDuration) : NaN;
```

Comment:
I think we should avoid explicitly setting values to `NaN`. Using `null` is functionally the same and raises the condition up to the type system.

Reply:
the other benefit of `NaN` instead of `null` is that usually, when doing math, it flows nicely within the code without the need to do a null check or something else. At the rendering, phase is usually ignored by the rendering environment (canvas, svg) or is just thrown as non-blocking error.